src/hc-scripts/download-cvs.ts

```diff
@@ -17,34 +19,25 @@ function downloadCv(application, destPath) {
     });
 
     request.on('error', error => {
-      fs.unlink(destPath, _ => undefined);
+      fs.unlink(destPath, () => undefined);
       reject(`Could not download CV for application ID ${application.id} (error ${error})`);
     });
   });
 }
 
-function downloadCvs(basePath) {
-  return getAllApplicationsWithTickets()
-    .then(applicationsWithTickets => {
-      const cvPromiseFunctions = applicationsWithTickets.map((application, i) => {
-        const destPath = path.resolve(process.cwd(), basePath, `Hack Cambridge CV ${i + 1}.pdf`);
-        return (() => downloadCv(application, destPath));
-      });
-
-      let promiseChain = Promise.resolve();
-
-      for (let i = 0; i < cvPromiseFunctions.length; i += simultaneousRequests) {
-        const promiseFunctionChunk = cvPromiseFunctions.slice(i, i + simultaneousRequests);
-        promiseChain = promiseChain.then(_ => {
-          console.log(`Downloading CVs ${i + 1} to ${i + promiseFunctionChunk.length}...`);
-          return Promise.all(promiseFunctionChunk.map(promiseFunc => promiseFunc()))
-            // tslint:disable-next-line:no-shadowed-variable
-            .then(_ => undefined);
-        });
-      }
-
-      return promiseChain;
-    });
+async function downloadCvs(basePath: string): Promise<void> {
+  const applicationsWithTickets = await getAllApplicationsWithTickets();
+  const shuffledApplications = _.shuffle(applicationsWithTickets);
+  const cvPromiseFunctions = shuffledApplications.map((application, i) => {
```

Comment:
Can we get rid of `applicationsWithTickets` and `shuffledApplications`? I think it could work with `then()` and method chaining, no?

Reply:
Alternatively we can just inline `applicationsWithTickets` e.g. `const shuffledApplications = _.shuffle(await getAllApplicationsWithTickets());`.  Generally `await` is preferred as it gives you direct style code.